paper-tabs.d.ts

```diff
@@ -0,0 +1,208 @@
+/// <reference path="../iron-menu-behavior/iron-menubar-behavior.d.ts" />
+/// <reference path="../iron-resizable-behavior/iron-resizable-behavior.d.ts" />
+/// <reference path="../paper-icon-button/paper-icon-button.d.ts" />
+/// <reference path="../paper-styles/color.d.ts" />
+/// <reference path="paper-tabs-icons.d.ts" />
+/// <reference path="paper-tab.d.ts" />
+
+/**
+ * Material design: [Tabs](https://www.google.com/design/spec/components/tabs.html)
+ *
+ * `paper-tabs` makes it easy to explore and switch between different views or functional aspects of
+ * an app, or to browse categorized data sets.
+ *
+ * Use `selected` property to get or set the selected tab.
+ *
+ * Example:
+ *
+ *     <paper-tabs selected="0">
+ *       <paper-tab>TAB 1</paper-tab>
+ *       <paper-tab>TAB 2</paper-tab>
+ *       <paper-tab>TAB 3</paper-tab>
+ *     </paper-tabs>
+ *
+ * See <a href="?active=paper-tab">paper-tab</a> for more information about
+ * `paper-tab`.
+ *
+ * A common usage for `paper-tabs` is to use it along with `iron-pages` to switch
+ * between different views.
+ *
+ *     <paper-tabs selected="{{selected}}">
+ *       <paper-tab>Tab 1</paper-tab>
+ *       <paper-tab>Tab 2</paper-tab>
+ *       <paper-tab>Tab 3</paper-tab>
+ *     </paper-tabs>
+ *
+ *     <iron-pages selected="{{selected}}">
+ *       <div>Page 1</div>
+ *       <div>Page 2</div>
+ *       <div>Page 3</div>
+ *     </iron-pages>
+ *
+ *
+ * To use links in tabs, add `link` attribute to `paper-tab` and put an `<a>`
+ * element in `paper-tab` with a `tabindex` of -1.
+ *
+ * Example:
+ *
+ * <pre><code>
+ * &lt;style is="custom-style">
+ *   .link {
+ *     &#64;apply --layout-horizontal;
+ *     &#64;apply --layout-center-center;
+ *   }
+ * &lt;/style>
+ *
+ * &lt;paper-tabs selected="0">
+ *   &lt;paper-tab link>
+ *     &lt;a href="#link1" class="link" tabindex="-1">TAB ONE&lt;/a>
+ *   &lt;/paper-tab>
+ *   &lt;paper-tab link>
+ *     &lt;a href="#link2" class="link" tabindex="-1">TAB TWO&lt;/a>
+ *   &lt;/paper-tab>
+ *   &lt;paper-tab link>
+ *     &lt;a href="#link3" class="link" tabindex="-1">TAB THREE&lt;/a>
+ *   &lt;/paper-tab>
+ * &lt;/paper-tabs>
+ * </code></pre>
+ *
+ * ### Styling
+ *
+ * The following custom properties and mixins are available for styling:
+ *
+ * Custom property | Description | Default
+ * ----------------|-------------|----------
+ * `--paper-tabs-selection-bar-color` | Color for the selection bar | `--paper-yellow-a100`
+ * `--paper-tabs-selection-bar` | Mixin applied to the selection bar | `{}`
+ * `--paper-tabs` | Mixin applied to the tabs | `{}`
+ * `--paper-tabs-content` | Mixin applied to the content container of tabs | `{}`
+ * `--paper-tabs-container` | Mixin applied to the layout container of tabs | `{}`
+ */
+interface PaperTabsElement extends Polymer.Element, Polymer.IronResizableBehavior, Polymer.IronMenubarBehavior {
+  selectable: string|null|undefined;
+  keyBindings: object;
```

Comment:
`keyBindings` seems like it should be [inherited from `iron-a11y-keys-behavior`](https://github.com/PolymerElements/iron-a11y-keys-behavior/blob/b6c34e5b7e1876bcf1a6b6b18915c83d43c8bcb9/iron-a11y-keys-behavior.d.ts#L72) rather than here.

Reply:
Since it's declared here, Analyzer emits it as a property. It doesn't know or care to look up to see if a property is also set by a behavior. I'm also not sure what the advantage would be of doing that. Actually, emitting the property means that TypeScript will verify that the type set here in the implementation matches the one in the behavior (which is actually why I had to add the `@type {!Object}` annotation in this PR).